apps/arkmarket/src/components/global-search.tsx

```diff
@@ -0,0 +1,191 @@
+      </div>
+    );
+  }
+
+  if (searchResults.data.length === 0) {
+    return (
+      <div className=" px-4 pb-4 pt-5">
+        <div className="mb-8 flex flex-col items-center gap-3 text-muted-foreground">
+          <Meh className="size-10" />
+          <p className="pl-3 text-xl font-semibold">
+            Sorry, there are no results for your search.
+          </p>
+        </div>
+
+        <GlobalSearchSuggestions onClose={onClose} />
+      </div>
+    );
+  }
+
+  return (
+    <CommandGroup forceMount className="px-4 pb-4 pt-5">
+      <p className="text-sm font-medium text-muted-foreground">Collections</p>
+      <div className="mt-4 flex flex-col gap-2.5">
+        {searchResults.data.map((searchResult) => {
+          return (
+            <CommandItem
+              value={`${searchResult.address}-${searchResult.name}`}
+              key={`${searchResult.address}-${searchResult.name}`}
+              className="flex cursor-pointer items-center gap-2 rounded-xs bg-transparent p-2 transition-colors"
+              asChild
+              onSelect={() => {
+                router.push(`/collection/${searchResult.address}`);
+                onClose();
+              }}
+            >
+              <Link
+                className={cn(focusableStyles)}
+                href={`/collection/${searchResult.address}`}
+                prefetch
+              >
+                <Media
+                  src={searchResult.image}
+                  alt={searchResult.name}
+                  height={64}
+                  width={64}
+                  className="size-8 rounded-xs"
+                />
+                <div>
+                  <div className="flex items-center gap-1">
+                    <p className="text-sm font-medium">{searchResult.name}</p>
+                    {searchResult.is_verified && (
+                      <VerifiedIcon className="size-3 text-background" />
+                    )}
+                  </div>
+                  <div className="flex items-center">
+                    <EthereumLogo2 className="-ml-1 size-4" />
+                    <p className="text-xs font-medium text-muted-foreground">
+                      {formatNumber(searchResult.token_count)} items
+                    </p>
+                  </div>
+                </div>
+              </Link>
+            </CommandItem>
+          );
+        })}
+      </div>
+    </CommandGroup>
+  );
+}
+
+export default function GlobalSearchWrapper() {
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [open, setOpen] = useState(false);
+  const [inputValue, setInputValue] = useState<string>("");
+  const [inputDebouncedValue, setInputDebouncedValue] = useDebounceValue(
+    "",
+    500,
+  );
+
+  const openSearch = () => setOpen(true);
+  const closeSearch = () => {
+    // Delay input close to let enough time to perform redirection if needed
+    void new Promise((resolve) => setTimeout(resolve, 200)).then(() =>
```

Comment:
Closing delay is too large, we need to find another solution to handle redirection issue.